poc_iot_injector/src/server.rs

```diff
@@ -139,85 +142,84 @@ async fn submit_txns(
-                                if let Err(err) = handle_txn_submission(
-                                    txn_details.clone(),
-                                    &mut shared_txn_service,
-                                )
-                                .await
-                                {
-                                    tracing::warn!("unable to submit_txn, {err:?}");
-                                } else {
-                                    tracing::info!("storing txn: {:?}", txn_details.hash_b64_url);
-                                    metrics::increment_counter!("poc_injector_receipt_count");
-                                    let _ = file_sink_write!(
-                                        "signed_poc_receipt_txn",
-                                        receipt_sender,
-                                        txn_details.txn
-                                    )
-                                    .await;
-                                }
-                            } else {
-                                tracing::info!("only storing txn: {:?}", txn_details.hash_b64_url);
-                                metrics::increment_counter!("poc_injector_receipt_count");
-                                let _ = file_sink_write!(
-                                    "signed_poc_receipt_txn",
-                                    receipt_sender,
-                                    txn_details.txn
-                                )
-                                .await;
-                            }
-                        }
-                        Err(err) => tracing::warn!("unable to construct txn_details, {err:?}"),
-                    },
+    let mut stream =
+        store.source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed());
+
+    let mut set = JoinSet::new();
+
+    while let Some(msg) = stream.next().await {
+        match msg {
+            Err(err) => tracing::warn!("skipping entry in stream: {err:?}"),
+            Ok(buf) => {
+                let shared_key = keypair.clone();
+                let mut shared_txn_service = txn_service.clone();
+                let receipt_sender_clone = receipt_sender.clone();
+
+                tracing::info!("Spawning submission tasks...");
+                set.spawn(async move {
+                    let _ = process_submission(
+                        buf,
+                        shared_key,
+                        do_submission,
+                        &receipt_sender_clone,
+                        &mut shared_txn_service,
+                    )
+                    .await;
+                });
+
+                if set.len() > MAX_CONCURRENT_SUBMISSIONS {
+                    tracing::info!("Processing {MAX_CONCURRENT_SUBMISSIONS} submissions");
+                    set.join_next().await;
                 }
             }
-        })
-        .await;
+        }
+    }
+
```

Comment:
Best to add the following here: 

```rust
while !set.is_empty() { 
     set.join_next().await;
 }
 ```
 
 `JoinSet`  aborts all of the threads it contains on drop

Reply:
Update [5069b47](https://github.com/helium/oracles/pull/258/commits/5069b47707ccfcbe6fec0de12a4714b312e9a738)